File: sq/SQ80Editor.swift

class SQ80Editor : ESQ1Editor {
  
  private static let _patchMap: [SynthPath:Sysexible.Type] = [
    [.global] : ChannelSettingsPatch.self,
    [.patch] : SQ80Patch.self,
    [.bank] : SQ80Bank.self,
  ]
  class override var patchMap: [SynthPath:Sysexible.Type] { return _patchMap }
}

File: sq/sq80_editor.js
const ESQ1Editor = require('./esq1_editor.js')
const Voice = require('./sq80_patch.js')

const editor = ESQ1Editor.editor
editor.trussMap = [
  ["global", 'channel'],
  ["patch", Voice.patchTruss],
  ["bank", Voice.bankTruss],
]

module.exports = {
  editor,
}